### DISCUSSION TITLE
Internal HW2 SNES - How to use controller port detection

### DISCUSSION THREAD

**Discussion by kaduleite** [darthcloud/BlueRetro#647] [Q&A]
How to use controller port detection on SNES?
Wiki Information says
> Option 1: Use existing system controller detection pin
> Only for Wii extension based system like NES & SNES classic

How to use this system controller detection pin?

**Comment by kaduleite** [comment]
After studying a lot, I understand how controller port detection works and I implemented successfully in Dreamcast.
But with SNES it really does not work.
### The blueretro is working fine, the issue is just the controller detection.
**My configuration:**
- SNES FAT
- ESP32 DEV Board
- Firmware HW2 v24.04 (tried with 1.92 - same problem)
- Controller port detection mode: switch button inside controller port (3v3 pull up resistor on IO35 and IO36)

**Synthoms**
I turn on the system and connect a bluetooth controller (I'm using a PS4 original controller). The controller works OK. If I plug a cabled controller the insertion is detected, controller led turns red, indicating that the controller corrected jumped to port 2, but the cable controller doesn't work at all.
The only way to use wired controllers is cutting ESP32 power.

**Questions**
1. Did I miss something?
2. Is it possible to use controller detection on SNES?

**Comment by Neorad1870** [comment]
Hello, you say that you put a resistor, between the two io35 and io36 or between two other locations? Thank you

**Reply by kaduleite** [reply]
Firs of all, sorry for delay.

A pull-up resistor is a resistor used in digital logic circuits to ensure a pin has a defined high logic level when no other component is actively driving it low. It's commonly used with switches and open-drain outputs to prevent floating inputs and ensure a predictable state. 
<img width="461" height="321" alt="image" src="https://github.com/user-attachments/assets/a57e5b17-4e38-4db9-95f7-8e2882d20cea" />

In this case it's used to assure you have low level on IO35 or IO36 when you plug the wired controller. In Dreamcast I did it work successfully, but in SNES it does not work. This is described in manual [topic 5 (Controller port detection), Option #3](https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-HW2-Internal-Install-Specification#5---controller-port-detection).

**Comment by Neorad1870** [comment]
Bonjour, Merci pour votre réponse , J’ai acheté un modèle snes interne sur AliExpress , ce n’est pas blueretro ils appelle ça bluemode , la détection des manettes fonctionnent, par contre je prend note pour la version dreamcast, merci beaucoup Cordialement João ***@***.***+41 76 405 17 36Le 22 juil. 2025 à 00:56, Kadu Leite ***@***.***> a écrit :﻿
Firs of all, sorry for delay.
A pull-up resistor is a resistor used in digital logic circuits to ensure a pin has a defined high logic level when no other component is actively driving it low. It's commonly used with switches and open-drain outputs to prevent floating inputs and ensure a predictable state.
image.png (view on web)
In this case it's used to assure you have low level on IO35 or IO36 when you plug the wired controller. In Dreamcast I did it work successfully, but in SNES it does not work. This is described in manual topic 5 (Controller port detection), Option #3.

**Reply by kaduleite** [reply]
Please, could you send me the link?

Kadu

**Comment by Neorad1870** [comment]
Bonjour, Voici le lien Je viens de trouver ça sur AliExpress : CHF46.02 | Adaptateur interne Bluetooth iBlueControlMod JP SFC pour Console EU SNES, contrôleur sans fil adapté pour PS3/PS4/PS5/Xbox One/Switchhttps://a.aliexpress.com/_EwSTqOwCordialement João ***@***.***+41 76 405 17 36Le 2 août 2025 à 19:14, Kadu Leite ***@***.***> a écrit :﻿
Please, could you send me the link?
Kadu

**Comment by kaduleite** [comment]
I had updated to 25.04, but it still does not work. Let's wait if in a future firmware @darthcloud solves this issue.
